IBM/carbon-addons-iot-react
# feat(table): Save view modal

Closes #1299

**Summary**
A highly customisable modal built for saving a table view. The component is built outside the actual table but only make sense in the context of saving a table view. A full example with this component and the table views will be completed as part of #1210. 

**Change List (commits, features, bugs, etc)**
- Created a TableSaveViewModal component that should be exported
- Created a TableSaveViewForm component as a subcomponent of the TableSaveViewModal
- Created 4 stories as a sub section of the table stories that demonstrates and with info explains the usage and how to customise the modal
- Wrote unit tests

**Acceptance Test (how to verify the PR)**
- Go to https://deploy-preview-1348--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table-tablesaveviewmodal--default-with-knobs
  - verify that all the knobs and actions works as expected
- Go to /https://deploy-preview-1348--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table-tablesaveviewmodal--prefilled 
  - make sure that the form is prefilled
- Go to https://deploy-preview-1348--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table-tablesaveviewmodal--custom-description-and-public-checkbox-using-overrides 
  - verify that the public checkbox is disabled and that the description is replaced by a working Accordion and that the story info makes sense
- Go to https://deploy-preview-1348--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table-tablesaveviewmodal--custom-form-using-overrides 
  - verify that the overridden form works and that the story info makes sense
